src/app/Delimiter.js

```diff
@@ -0,0 +1,136 @@
+// @ts-check
+
+import {
+  filterEmpty,
+  findCapturedValue,
+  isMatch,
+  isStartsWith,
+  shallowCopy,
+} from '../lib/utils.js';
+
+class Delimiter {
+  /** @type {Array<string>} */
+  #defaultDelimiters = [',', ':'];
+
+  /** @type {object} */
+  #customDelimiterFormatRegEx = /\/\/.*?\\n/;
+
+  /** @type {object} */
+  #customDelimiterRegEx = /\/\/(.*?)\\n/;
+
+  /** @type {Array<string>} */
+  #customDelimiterMatchers = ['//', '\\n'];
+
+  /**
+   *
+   * @param {string} value
+   * @returns {boolean}
+   */
+  hasCustomDelimiter(value) {
+    return (
+      isStartsWith(this.#customDelimiterMatchers[0], value) &&
+      isMatch(this.#customDelimiterFormatRegEx, value)
+    );
+  }
+
+  /**
+   *
+   * @param {string} value
+   * @returns {string | null}
+   */
+  getCustomDelimiter(value) {
+    return findCapturedValue(this.#customDelimiterRegEx, value);
+  }
+
+  /**
+   *
+   * @param {string} value
+   * @returns {{ start: number; end: number }}
+   */
+  #getCustomDelimiterMatcherLocation(value) {
+    return {
+      start: value.indexOf(this.#customDelimiterMatchers[0]),
+      end: value.indexOf(this.#customDelimiterMatchers[1]),
+    };
+  }
+
+  /**
+   *
+   * @returns {{ start: number; end: number; }}
+   */
+  #getCustomDelimiterMatcherLength() {
+    return {
+      start: this.#customDelimiterMatchers[0].length,
+      end: this.#customDelimiterMatchers[1].length,
+    };
+  }
+
+  /**
+   *
+   * @param {string} value
+   * @returns {string}
+   */
+  #removeCustomDelimiterFormat(value) {
+    return value.slice(
+      this.#getCustomDelimiterMatcherLocation(value).end +
+        this.#getCustomDelimiterMatcherLength().end,
+    );
+  }
+
+  /**
+   *
+   * @param {string} value
+   * @returns {Array<string>}
+   */
+  #getDelimiters(value) {
+    const delimiters = shallowCopy(this.#defaultDelimiters);
+
+    if (this.hasCustomDelimiter(value)) {
+      delimiters.push(this.getCustomDelimiter(value));
+    }
+
+    return delimiters;
+  }
+
+  /**
+   *
+   * @param {string} value
+   * @returns {object}
+   */
+  #getDelimiterRegEx(value) {
+    return new RegExp(`[${this.#getDelimiters(value).join('')}]`);
+  }
```

Comment:
지금은 크게 신경 쓸 부분은 아니지만 계속해서 새로운 RegExp를 생성하는 것 보다 캐싱하면 조금 더 아낄 수 있지 않을까? 하는 생각이 들었어요

```suggestion
#getDelimiterRegEx(value) {
  if (!this.cachedRegEx || this.lastValue !== value) {
    this.cachedRegEx = new RegExp(`[${this.#getDelimiters(value).join('')}]`);
    this.lastValue = value;
  }
  return this.cachedRegEx;
}
```

이런 방법은 어떨까요?

Reply:
오 너무 좋은거 같아요. 

구현하면서 성능과 관련해서는 신경쓰지 못한 부분이 많았는데 이런 부분 너무 좋은거 같아요!

감사합니다 :)